datafusion/core/tests/sql/timestamp.rs

```diff
@@ -1070,15 +1075,19 @@ async fn cast_string_to_time() {
     let result = try_execute_to_batches(&ctx, sql).await;
     assert_eq!(
         result.err().unwrap().to_string(),
-        "Arrow error: Cast error: Cannot cast string 'not a time' to value of Time64(Nanosecond) type"
+        "simplify_expressions\ncaused by\nInternal error: Optimizer rule 'simplify_expressions' failed due to unexpected error: \
+        Arrow error: Cast error: Cannot cast string 'not a time' to value of Time64(Nanosecond) type. \
+        This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker"
```

Comment:
Ditto. We should find a way to update the error message when the error is expected. However, this is not a big deal now, because the default value of `skip_failed_rules` is `false` and customers won't see this error message.